comfy_cli/utils.py

```diff
@@ -78,3 +95,23 @@ def f(incomplete: str) -> list[str]:
         return [opt for opt in opts if opt.startswith(incomplete)]
 
     return f
+
+def download_progress(url, fname, cwd=".", allow_redirects=True):
+    """download url to local file fname and show a progress bar.
+    See https://stackoverflow.com/q/37573483"""
```

Comment:
Typer already has a progress bar I believe: 

https://typer.tiangolo.com/tutorial/progressbar/

Reply:
hmm, this application of the `tqdm` progress bar turns out to be particularly cursed to replace with the one from `rich`. And technically `tqdm` isn't a new requirement since it's also needed by comfyui core. But I'll give replacing it a shot.

Reply:
done! Turned out to be easier than I thought it would be